apps/tangle-dapp/app/bridge/ChainSelectors.tsx

```diff
@@ -0,0 +1,161 @@
+'use client';
+
+import { DropdownMenuTrigger as DropdownTrigger } from '@radix-ui/react-dropdown-menu';
+import { ChainConfig } from '@webb-tools/dapp-config/chains/chain-config.interface';
+import { ArrowRight } from '@webb-tools/icons/ArrowRight';
+import { ChainIcon } from '@webb-tools/icons/ChainIcon';
+import ChainButton from '@webb-tools/webb-ui-components/components/buttons/ChainButton';
+import {
+  Dropdown,
+  DropdownBody,
+} from '@webb-tools/webb-ui-components/components/Dropdown';
+import { MenuItem } from '@webb-tools/webb-ui-components/components/MenuItem';
+import { ScrollArea } from '@webb-tools/webb-ui-components/components/ScrollArea';
+import { FC, useCallback } from 'react';
+
+import { useBridge } from '../../context/BridgeContext';
+
+interface ChainSelectorProps {
+  title: string;
+  selectedChain: ChainConfig | null;
+  chainOptions: ChainConfig[];
+  onSelectChain: (chain: ChainConfig) => void;
+  className?: string;
+}
+
+const ChainSelectors: FC = () => {
+  const {
+    sourceChain,
+    setSourceChain,
+    destinationChain,
+    setDestinationChain,
+    supportedSourceChains,
+    supportedDestinationChains,
+  } = useBridge();
+
+  const onSetSourceChain = useCallback(
+    (chain: ChainConfig) => {
+      setSourceChain(chain);
+      // If the source chain is the same as the destination chain, clear the destination chain.
+      if (chain.id === destinationChain?.id) {
+        setDestinationChain(null);
+      }
+    },
+    [destinationChain?.id, setDestinationChain, setSourceChain]
+  );
+
+  const onSetDestinationChain = useCallback(
+    (chain: ChainConfig) => {
+      setDestinationChain(chain);
+      // If the destination chain is the same as the source chain, clear the source chain.
```

Comment:
It seems that instead of switching and nullifying things that we should not present the same chain as an option? Or do you think this is better. I'm ambivalent is this is the logic you think is worth having.

Reply:
Working on it right now!